app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js

```diff
@@ -0,0 +1,283 @@
+import nock from 'nock';
+import { ClientFunction, Selector } from 'testcafe';
+import { extractInnerText } from 'buying-catalogue-library';
+import { orderApiUrl } from '../../../../../../../../config';
+import content from '../manifest.json';
+import { nockAndErrorCheck, setState, authTokenInSession } from '../../../../../../../../test-utils/uiTestHelper';
+import { sessionKeys } from '../../../../../../../../helpers/routes/sessionHelper';
+
+const pageUrl = 'http://localhost:1234/order/organisation/order-id/catalogue-solutions/1';
+
+const getLocation = ClientFunction(() => document.location.href);
+
+const orderItem = {
+  serviceRecipient: {
+    odsCode: 'OX3',
+    name: 'Some service recipient 2',
+  },
+  catalogueItemType: 'Solution',
+  catalogueItemName: 'Some item name',
+  catalogueItemId: '10000-001',
+  quantity: 3,
+  estimationPeriod: 'year',
+  provisioningType: 'ondemand',
+  type: 'flat',
+  currencyCode: 'GBP',
+  itemUnit: {
+    name: 'patient',
+    description: 'per consultation',
```

Comment:
As with the general UI test, it would be better to use an example of an on-demand price here; an option might be consultations, for example: 

```javascript
itemUnit: {
    name: 'consultationCore',
    description: 'per consultation – core hours',
}
```

Reply:
Done